### PR TITLE
fix(plugin-coverage): merge multiple results for a file

### LINKED ISSUE #560: Coverage has significantly decreased after Nx update
### What happened?

The code coverage score has decreased by 30 points after Nx update.

### What would you expect to happen?

The score would correctly update to the changes (no big fluctuation was expected).

### Code PushUp package version

0.25.4

### What operation system are you on?

Linux

### Node version

20.0.0

### Relevant information
- [commit link](https://github.com/code-pushup/cli/commit/b2c87eb716a84f82b5d59e7238ffcae7147f707a)
- [portal link](https://quality-metrics-staging.web.app/portal/code-pushup/cli/commit/b2c87eb716a84f82b5d59e7238ffcae7147f707a/dashboard)

The coverage is lower when running the plugin locally as well.
image

### PR DESCRIPTION
**Pull request by Tlacenka**
Closes #560 
Fixes part of #647 

In this PR, I implement a merging logic in order to collect coverage for the same file system from multiple test suites:
- In case different coverage runs identify different branches of code/functions, all that are identified by any tool are registered.
- The hits of functions, branches and lines are merged. As long as a function, branch or line is covered by at least one test suite, it is considered covered.

Tested on our CLI + unit tested.

### CODE REVIEW

**Review by BioPhoton — APPROVED** [review]
LGTM

File: packages/plugin-coverage/src/lib/runner/lcov/merge-lcov.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        JSON.stringify({ line, block, branch }),
+      ),
+    ),
+  ].map(
+    functionJSON =>
+      JSON.parse(functionJSON) as {
+        line: number;
+        block: number;
+        branch: number;
+      },
```
Comment (matejchalk) [inline packages/plugin-coverage/src/lib/runner/lcov/merge-lcov.ts:140]: This way the object type is derived from the original source, which makes it a bit safer:

```suggestion
      JSON.parse(functionJSON) as Pick<BranchesDetails, 'line' | 'block' | 'branch'>,
```

**Review by matejchalk — APPROVED** [review]
Great effort :+1:

File: packages/plugin-coverage/src/lib/runner/lcov/merge-lcov.ts (hunk 1 above)
Comment (Tlacenka) [reply packages/plugin-coverage/src/lib/runner/lcov/merge-lcov.ts:140]: Good point!

**Review by matejchalk — DISMISSED** [review]

**Review by matejchalk — APPROVED** [review]

**Pull request merged**